arviz/sel_utils.py

```diff
@@ -180,14 +160,22 @@ def xarray_var_iter(data, var_names=None, combined=False, skip_dims=None, revers
     if var_names is None and isinstance(data, xr.DataArray):
         data_to_sel = {data.name: data}
 
+    if isinstance(dim_order, str):
+        dim_order = [dim_order]
+
     for var_name, selection, iselection in xarray_sel_iter(
         data,
         var_names=var_names,
         combined=combined,
         skip_dims=skip_dims,
         reverse_selections=reverse_selections,
     ):
-        yield var_name, selection, iselection, data_to_sel[var_name].sel(**selection).values
+        selected_data = data_to_sel[var_name].sel(**selection)
+        if dim_order is not None:
+            dim_order_selected = [dim for dim in dim_order if dim in selected_data.dims]
+            if dim_order_selected:
+                selected_data = selected_data.transpose(*dim_order, ...)
```

Comment:
```suggestion
                selected_data = selected_data.transpose(*dim_order_selected, ...)
```

should this be like this? isn't it possible that dim_order has dimensions not present in the data otherwise?

Reply:
Yes